src/daft-core/src/series/ops/arithmetic.rs

```diff
@@ -211,6 +211,24 @@ impl Rem for &Series {
         }
     }
 }
+
+impl Series {
+    pub fn floor_div(&self, rhs: &Self) -> DaftResult<Self> {
+        let output_type = InferDataType::from(self.data_type())
+            .floor_div(&InferDataType::from(rhs.data_type()))?;
+        let lhs = self;
+        match &output_type {
+            #[cfg(feature = "python")]
+            DataType::Python => run_python_binary_operator_fn(lhs, rhs, "floordiv"),
+            output_type if output_type.is_numeric() => {
+                let div_floor = lhs.div(rhs)?.floor()?;
+                div_floor.cast(output_type)
+            }
```

Comment:
You can optimize this if output type is an integer by operating on the integers themselves. `lhs.div(rhs)?.floor()?` will actually cast both sides to float64 then doing the div, but we shouldn't need to do this if both sides are integers.

Rough example:
```
output_type if output_type.is_integer() => {
                with_match_integer_daft_types!(output_type, |$T| {
                    Ok(cast_downcast_op!(lhs, rhs, output_type, <$T as DaftDataType>::ArrayType, div)?.into_series())
                })
            }
```

Reply:
Note that you will have to modify `daft-core/src/array/ops/arithmetic.rs` to implement `Div` for all integer types. I recommend looking at the rem implementation https://github.com/Eventual-Inc/Daft/blob/main/src/daft-core/src/array/ops/arithmetic.rs#L150 . Lmk if you need help with it.

Reply:
Updated based on your suggestions.